packages/server/modules/core/tests/streams.spec.ts

```diff
@@ -215,6 +219,35 @@ describe('Streams @core-streams', () => {
       const userIsCollaborator = await isStreamCollaborator(userTwo.id, streamId)
       expect(userIsCollaborator).to.not.be.ok
     })
+    it('Server guests cannot be stream owners', async () => {
+      const guestGuy: BasicTestUser = {
+        name: 'Some we do not fully trust',
+        email: 'shady@contractor.company',
+        password: 'foobar123',
+        id: ''
+      }
+
+      await createTestUsers([guestGuy])
+
+      await changeUserRole({
+        userId: guestGuy.id,
+        role: Roles.Server.Guest,
+        guestRoleEnabled: true
+      })
+
+      await addOrUpdateStreamCollaborator(
+        testStream.id,
+        guestGuy.id,
+        Roles.Stream.Owner,
+        userOne.id
+      )
+        .then(() => {
```

Comment:
The promise is awaited anyway, so instead of `then()`/`catch()` you can convert this to a standard try-catch

Reply:
i always get bamboozled by this. Not having a nice way of testing for a promise rejection is grinding my gears.

Reply:
> Not having a nice way of testing for a promise rejection is grinding my gears.

What do you mean by this? You should be able to convert that into this:

```ts
try {
   await addOrUpdateStreamCollaborator(
        testStream.id,
        guestGuy.id,
        Roles.Stream.Owner,
        userOne.id
      )
    throw new Error('This should have thrown')
} catch (e) {
  expect(err.message).to.include('Server guests cannot own streams')
}
```

Reply:
i mean in automate we have [this](https://github.com/specklesystems/speckle-automate/blob/519e4edcecdd51ad21b76285a0470020855af0d1/components/automate-app/tests/unit/authz/pipeline.spec.ts#L85) which is super nice

but ofcourse i can convert this into async / await